### ISSUE TITLE
GENERAL: Allow all serial-monitor based tools to communicate to xemu over tcp port 4510

### ISSUE THREAD

**Issue by gurcei** [MEGA65/mega65-tools#32]
Based on this Discord conversation:

- https://discord.com/channels/719326990221574164/781481205639020554/805593238668181545

It got me thinking it'd be nice to get these mega65-tools talking to xemu over its tcp 4510 debug port.

I had made some steps in that direction with m65dbg already, so just a matter of porting those ideas across, perhaps into m65common.

**Comment by gurcei** [comment]
Had a little sniff at this now.

The efforts I made to support this in m65dbg were a little easier to do, as I've typically been using/compiling m65dbg under cygwin, where I've got common network libraries to what linux and macs have.

To get this working with Windows mingw64, it'll be a different kettle of fish though, need to get acquainted with the windows way doing doing these things (need to use winsock2.h?)

**Comment by gurcei** [comment]
Appears to be working well when I tested running xemu from my `220-uartmon-update` branch against m65.exe for this example case:

Xemu:
`build/bin/xmega65.native -uartmon :4510`

M65:
`bin/m65.exe -l tcp -S`

This created a somewhat successful screenshot of the xemu screen:
![xemu m65 screenshot](http://gurce.net/Logs/images/img_2021_05_18_6_02_16.png)

I also tried running `bin/mega65_ftp.exe -l tcp`. This managed to do the `GO64`, but stalled after that.

So while the conversation with xemu doesn't work as well as with real hardware, this appears to be due to xemu presently not supporting some of the io-registers used by tools like m65 to perform their remote-io tricks.

See this discord thread for further discussion on that:
- https://discord.com/channels/719326990221574164/805492126128930846/844189640374222888

So given as the goal of the card was to provide tcp connection capabilities, I believe this goal has been achieved.

For xemu-specific matters on missing io-registers, we can raise cards on xemu to sort these matters out later
